### PR TITLE
convert code-blocks to runnableExamples in io

### PR DESCRIPTION
**Pull request by vercingetorx**

### CODE REVIEW

File: lib/system/io.nim [hunk 1]
```diff
@@ -912,14 +912,14 @@ iterator lines*(filename: string): string {.tags: [ReadIOEffect].} =
...
-  ##       buffer.add(line.replace("a", "0") & '\n')
-  ##     writeFile(filename, buffer)
+  runnableExamples:
+     import std/strutils
+  
+     proc transformLetters(filename: string) =
+       var buffer = ""
+       for line in filename.lines:
+         buffer.add(line.replace("a", "0") & '\n')
+       writeFile(filename, buffer)
```
Comment (konsumlamm) [inline lib/system/io.nim]:
```suggestion
    import std/strutils

    proc transformLetters(filename: string) =
      var buffer = ""
      for line in filename.lines:
        buffer.add(line.replace("a", "0") & '\n')
      writeFile(filename, buffer)
```
Use 2 spaces to indent, not 3. Ditto for the other example.

File: lib/system/io.nim [hunk 2]
```diff
@@ -931,14 +931,13 @@ iterator lines*(f: File): string {.tags: [ReadIOEffect].} =
...
-  ##         if letter == '0':
-  ##           result.zeros += 1
-  ##       result.lines += 1
+  runnableExamples:
+     proc countZeros(filename: File): tuple[lines, zeros: int] =
+       for line in filename.lines:
+         for letter in line:
+           if letter == '0':
+             result.zeros += 1
+         result.lines += 1
```
Comment (konsumlamm) [inline lib/system/io.nim]:
```suggestion
  runnableExamples:
    proc countZeros(filename: File): tuple[lines, zeros: int] =
      for line in filename.lines:
        for letter in line:
          if letter == '0':
            result.zeros += 1
        result.lines += 1
```
When I say "ditto for the other example", I'm not just saying that for fun :P

Comment (vercingetorx) [reply lib/system/io.nim]: Oh I missed that, just woke up.

**Comment by vercingetorx** [comment]
As I understand it, going through and converting code-blocks to runnable examples is something that needs to be done. If so, that's something I can continue to work on.

**Review by ringabout — APPROVED** [review]

**Review by timotheecour — APPROVED** [review]

**Comment by ringabout** [comment]
Maybe you can handle more files per time(For example convert 5 ~ 10 files in one PR). And there are also other forms of docs improvement(double backticks => single backticks and converts to `.. warning::`, `..note::` etc.) if you would like to.

**Comment by timotheecour** [comment]
@xioren 
> As I understand it, going through and converting code-blocks to runnable examples is something that needs to be done. If so, that's something I can continue to work on.

yes, please!
and, exactly as @xflywind said, you can bundle several related cleanups in the same PR (ok to start with fewer until you feel comfortable)

see @konsumlamm 's PRs for a good reference of what such cleanups can be: https://github.com/nim-lang/Nim/pulls?q=is%3Apr+sort%3Aupdated-desc+author%3Akonsumlamm+is%3Amerged

see also https://github.com/timotheecour/Nim/issues/534 (common review comments)

**Pull request merged**